File: JavaScript/BigO/rule-2.js

//TODO:  REMOVE CONSTANTS

// remover las constantes al caluclar Big O 

const printFirstItemThenFirstHalfThenSayHi100Times = (items) =>{
    
    
    middleIndex = Math.floor(items.length/2);
    index = 0;

    while ( index < middleIndex){
        console.log(items[index]);
        index++;
    }

    for (let i = 0; i < 100; i++) {
        console.log("Hi");
    }
}


printFirstItemThenFirstHalfThenSayHi100Times([1,2,3,4,5])

//O(2n) pero eliminamos las constantes
// O(n)

File: Java/src/BigO/Rule2.java
package BigO;

public class Rule2 {

    public static void main(String[] args) {


    }

    public static void printFirstItemThenFirstHalfThenSayHi100Times(String[]items){
        int middleIndex = items.length/2;
        int index = 0;

        while ( index < middleIndex){
            System.out.println(items[index]);
            index++;
        }

        for (int i = 0; i < 100; i++) {
            System.out.println("Hi");
        }
    }
}
//O(2n) pero eliminamos las constantes
// O(n)